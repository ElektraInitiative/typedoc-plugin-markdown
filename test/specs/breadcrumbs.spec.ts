import * as Handlebars from 'handlebars';
import { ProjectReflection, Reflection } from 'typedoc';

describe(`Breadcrumbs:`, () => {
  let moduleReflection: Reflection;
  let classReflection: Reflection;

  describe(`(with readme)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['breadcrumbs.ts']);
      moduleReflection = (project.children as any)[0];
      classReflection = project.getChildByName('Breadcrumbs') as Reflection;
    });

    test(`should compile README breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: project,
          url: 'README.md',
        }),
      ).toMatchSnapshot();
    });

    test(`should compile entryPoint (globals) breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: project,
          url: 'globals.md',
        }),
      ).toMatchSnapshot();
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: moduleReflection,
          url: moduleReflection.url,
        }),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: classReflection,
          url: classReflection.url,
        }),
      ).toMatchSnapshot();
    });
  });
  describe(`(without readme)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['breadcrumbs.ts'], { readme: 'none' });
      moduleReflection = (project.children as any)[0];
      classReflection = project.getChildByName('Breadcrumbs') as Reflection;
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: moduleReflection,
          url: moduleReflection.url,
        }),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: project,
          model: classReflection,
          url: classReflection.url,
        }),
      ).toMatchSnapshot();
    });
  });
});
