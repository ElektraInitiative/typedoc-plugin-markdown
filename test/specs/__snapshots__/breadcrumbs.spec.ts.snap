// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs: (with readme) should compile README breadcrumbs' 1`] = `"typedoc-plugin-markdown / [Exports](../modules.md)"`;

exports[`Breadcrumbs: (with readme) should compile class breadcrumbs' 1`] = `"[typedoc-plugin-markdown](../README.md) / [Exports](../modules.md) / Breadcrumbs"`;

exports[`Breadcrumbs: (with readme) should compile entryPoint (globals) breadcrumbs' 1`] = `"[typedoc-plugin-markdown](../README.md) / [Exports](../modules.md)"`;

exports[`Breadcrumbs: (with readme) should compile module breadcrumbs' 1`] = `"[typedoc-plugin-markdown](../README.md) / [Exports](../modules.md) / Breadcrumbs"`;

exports[`Breadcrumbs: (without readme) should compile class breadcrumbs' 1`] = `"[typedoc-plugin-markdown](../README.md) / Breadcrumbs"`;

exports[`Breadcrumbs: (without readme) should compile module breadcrumbs' 1`] = `"[typedoc-plugin-markdown](../README.md) / Breadcrumbs"`;
